src/axom/sidre/core/MFEMSidreDataCollection.cpp

```diff
@@ -2399,119 +2418,127 @@ void MFEMSidreDataCollection::reconstructMesh()
-        }
+      {
+        m_fespaces[basis_name] = std::unique_ptr<mfem::FiniteElementSpace>(
+          new mfem::FiniteElementSpace(mesh,
+                                       m_fecolls.at(basis_name).get(),
+                                       vdim,
+                                       ordering));
       }
+    }
 
-      double* values = value_view->getData();
+    double* values = value_view->getData();
 
-      if(is_gridfunc)
-      {
+    if(is_gridfunc)
+    {
   #if defined(AXOM_USE_MPI) && defined(MFEM_USE_MPI)
-        auto parfes = dynamic_cast<mfem::ParFiniteElementSpace*>(
-          m_fespaces.at(basis_name).get());
-        if(parfes)
-        {
-          m_owned_gridfuncs.emplace_back(
-            new mfem::ParGridFunction(parfes, values));
-        }
-        else
-  #endif
-        {
-          m_owned_gridfuncs.emplace_back(
-            new mfem::GridFunction(m_fespaces.at(basis_name).get(), values));
-        }
-
-        // Register a non-owning pointer with the base subobject
-        DataCollection::RegisterField(field_grp->getName(),
-                                      m_owned_gridfuncs.back().get());
+      auto parfes = dynamic_cast<mfem::ParFiniteElementSpace*>(
+        m_fespaces.at(basis_name).get());
+      if(parfes)
+      {
+        m_owned_gridfuncs.emplace_back(new mfem::ParGridFunction(parfes, values));
       }
       else
+  #endif
       {
-        m_owned_quadfuncs.emplace_back(
-          new mfem::QuadratureFunction(m_quadspaces.at(basis_name).get(),
-                                       values,
-                                       vdim));
-        // Register a non-owning pointer with the base subobject
-        DataCollection::RegisterQField(field_grp->getName(),
-                                       m_owned_quadfuncs.back().get());
+        m_owned_gridfuncs.emplace_back(
+          new mfem::GridFunction(m_fespaces.at(basis_name).get(), values));
       }
+
+      // Register a non-owning pointer with the base subobject
+      DataCollection::RegisterField(field_grp->getName(),
+                                    m_owned_gridfuncs.back().get());
+    }
+    else
+    {
+      m_owned_quadfuncs.emplace_back(
+        new mfem::QuadratureFunction(m_quadspaces.at(basis_name).get(),
+                                     values,
+                                     vdim));
+      // Register a non-owning pointer with the base subobject
+      DataCollection::RegisterQField(field_grp->getName(),
+                                     m_owned_quadfuncs.back().get());
+    }
+  }
+}
+
+void MFEMSidreDataCollection::reconstructFields()
+{
+  sidre::Group* f = m_bp_grp->getGroup("fields");
+  for(auto idx = f->getFirstValidGroupIndex(); sidre::indexIsValid(idx);
+      idx = f->getNextValidGroupIndex(idx))
+  {
+    Group* field_grp = f->getGroup(idx);
+    // The nodal grid function will already have been reconstructed
+    if(field_grp->getName() != m_meshNodesGFName)
```

Comment:
I wonder if there should still be a check since one can't guarantee the order that users might call this.

Reply:
I may be forgetting something but I think we should be able to guarantee the order since these methods are private and only accessible through `UpdateMeshAndFieldsFromDS` which does everything in the correct order